utils/docker/Dockerfile.centos.7

```
@@ -164,10 +150,15 @@ COPY utils/sl utils/sl
 COPY utils/build.config utils/
 RUN chown -R daos_server.daos_server /home/daos
 USER daos_server:daos_server
-RUN ln -s utils/sl site_scons
 
-RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || \
-    (scons --build-deps=yes install PREFIX=/opt/daos --deps-only && \
+# Control what to build.  By default Dockerfiles build everything to allow for
+# ease-of-use for users, however in CI everything is turned off and then
+# selectively enabled.  Turning off any step disables all future steps.
+ARG DAOS_DEPS_BUILD=yes
+ARG DEPS_JOBS=1
+
+RUN ln -s utils/sl site_scons && [ "$DAOS_DEPS_BUILD" != "yes" ] || \
```

Use the more natural boolean logic available in the Bourne shell, and avoid double negatives:

```suggestion
RUN ln -s utils/sl site_scons && "$DAOS_DEPS_BUILD" && \
```

---

I wish we could, but each build step has to return true of the build stops, so you end up with quite complex syntax unfortunatly.

---

Ahh.  So then you can use:
```suggestion
RUN ln -s utils/sl site_scons && if "$DAOS_DEPS_BUILD"; then \
```
obviously with a `fi` and the end of the block given that:

```
$ (set -x; true && false && true; echo $?)
+ true
+ false
+ echo 1
1
$ (set -x; true && if false; then true; fi; echo $?)
+ true
+ false
+ echo 0
0
```

and get even more readable code (IMHO).

---

It was doing this until recently when I changed it to use the || trick as I thought it was shorter and therefore easier to read.

I could back out this change to put the ln the same RUN command as the build, I'm going to try and rename utils/sl to site_scons soon anyway so it'll hopefully be going away soon.